### PR TITLE
Mam ch ep announcements

### PR DESCRIPTION
**Pull request by tsvisabo**
- Add convenience functions (api/examples) to announce ch/ep and send on ch/ep
- replace old usage
- some documentation

### CODE REVIEW

File: mam/api/tests/test_api.c [hunk 1]
```diff
@@ -240,23 +240,26 @@ static void test_api_write_header(
...
     TEST_ASSERT(mam_psk_t_set_add(&psks, pska) == RC_OK);
     TEST_ASSERT(mam_psk_t_set_add(&psks, pskb) == RC_OK);
   } else if (mam_msg_keyload_ntru == keyload) {
     TEST_ASSERT(mam_ntru_pk_t_set_add(&ntru_pks, ntru_pk) == RC_OK);
   }
 
-  TEST_ASSERT(mam_api_bundle_write_header(api, ch, ep, ch1, ep1, psks, ntru_pks,
-                                          0, bundle, msg_id) == RC_OK);
+  if (mam_msg_pubkey_epid == pubkey) {
+    TEST_ASSERT(mam_api_bundle_write_header_on_channel(
```
Comment (thibault-martinez) [inline mam/api/tests/test_api.c]: Should be endpoint here ?

File: mam/api/tests/test_api.c [hunk 2]
```diff
@@ -240,23 +240,26 @@ static void test_api_write_header(
...
+                    api, ch, psks, ntru_pks, 0, bundle, msg_id) == RC_OK);
+  } else if (mam_msg_pubkey_chid1 == pubkey) {
+    TEST_ASSERT(mam_api_bundle_write_header_announce_new_channel(
+                    api, ch, ch1a, psks, ntru_pks, 0, bundle, msg_id) == RC_OK);
+  } else if (mam_msg_pubkey_epid1 == pubkey) {
+    TEST_ASSERT(mam_api_bundle_write_header_announce_new_endpoint(
+                    api, ch, ep1a, psks, ntru_pks, 0, bundle, msg_id) == RC_OK);
+  } else {
+    TEST_ASSERT(mam_api_bundle_write_header_on_endpoint(
```
Comment (thibault-martinez) [inline mam/api/tests/test_api.c]: Should be channel here ?

File: mam/api/api.c [hunk 3]
```diff
@@ -102,6 +102,34 @@ static mam_channel_t *mam_api_get_channel(mam_api_t const *const api,
   return NULL;
 }
 
+mam_endpoint_t *mam_api_get_endpoint(mam_api_t const *const api,
```
Comment (thibault-martinez) [inline mam/api/api.c:105]: Do you think the overhead of doing this without the associated `channel_id ` is  too big ?

File: mam/examples/send-header.c [hunk 4]
```diff
@@ -31,6 +36,17 @@ int main(int ac, char **av) {
     }
   }
 
+  msg_pubkey_int = atoi(av[4]);
+  if (msg_pubkey_int < 0 || msg_pubkey_int > 4) {
```
Comment (thibault-martinez) [inline mam/examples/send-header.c]: Should be > 3 ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: mam/api/api.c (hunk 3 above)
Comment (tsvisabo) [reply mam/api/api.c:105]: obviously depends on the number of channels, but there's no reason why user should not be able to provide the channel

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**